File: ios/Models/MotionBlurConfiguration.swift
//
//  MotionBlurConfiguration.swift
//  Pods
//
//  Created by rit3zh CX on 9/23/26.
//

import CoreGraphics

struct MotionBlurConfiguration: Equatable {
  static let `default` = MotionBlurConfiguration()
  static let sampleCountRange = 4...64
  var intensity: CGFloat = 40
  var speedForMaxBlur: CGFloat = 1500
  var maxSampleCount = 32
  var isEnabled = true
  var isLive = false
  var exposure: CGFloat {
    speedForMaxBlur > 0 ? intensity / speedForMaxBlur : 0
  }
}

File: android/src/main/java/expo/modules/motionblurview/models/MotionBlurConfiguration.kt
package expo.modules.motionblurview.models

internal data class MotionBlurConfiguration(
  val intensity: Double = 40.0,
  val speedForMaxBlur: Double = 1500.0,
  val maxSampleCount: Int = 32,
  val isEnabled: Boolean = true
) {
  val exposure: Double
    get() = if (speedForMaxBlur > 0) intensity / speedForMaxBlur else 0.0

  companion object {
    val DEFAULT = MotionBlurConfiguration()
    val SAMPLE_COUNT_RANGE = 4..64
  }
}
